app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/login/LoginViewModel.kt

```diff
@@ -1,6 +1,79 @@
 package com.github.alfabravo2013.readyforexams.presentation.login
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.util.OneTimeEvent
+import com.github.alfabravo2013.readyforexams.util.isInvalidEmail
+import com.github.alfabravo2013.readyforexams.util.isInvalidPassword
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 
 class LoginViewModel : ViewModel() {
+    private var emailAddress = ""
```

Comment:
Ideally we can move this emailAddress and password to a LocalDataSource. The reason would be that once we implement backend for authentication is going to be easier to replace.
If I will get access to it I'll create the flow myself and you will have it as an example.